test/e2e/upgrade_management_cluster_test.go

```diff
@@ -0,0 +1,352 @@
+package e2e
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"html/template"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	bmo "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	clusterv1old "sigs.k8s.io/cluster-api/api/v1alpha3"
```

Comment:
I think it would be better to spell out the version here.
```suggestion
	clusterv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
```
Or maybe
```suggestion
	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
```

Now that I'm thinking about it, why is it v1alpha3? Are we not going from v1alpha4 -> v1beta1?

Reply:
done

Reply:
PS: Changing that to capiv1alpha4 caused the error `no kind is registered for the type v1alpha4.MachineList in scheme "pkg/runtime/scheme.go:100"` so I added it in suite file to fix that !  please take a look :)

Reply:
Good! I'm not sure if there are other implications from this `AddToScheme`. It could be that we should only us it for upgrade tests, but I'm not sure about it. Maybe someone else knows?

Reply:
@lentzi90 I do not see it being set to `sigs.k8s.io/cluster-api/api/v1alpha3` anymore